src/crypto/shelley/transactions/utils.js

```diff
@@ -27,19 +43,119 @@ export const getTxOutputTotal = async (
     const value = new BigNumber(await (await output.value()).to_str())
     sum = sum.plus(value)
   }
+  if (shift) {
+    return sum.shiftedBy(-CONFIG.DECIMAL_PLACES_IN_ADA)
+  }
   return sum
 }
 
-// TODO: test
-export const v2SkKeyToV3Key = async (v2Key: HdWallet.XPrv): PrivateKey => {
-  return await PrivateKey.from_extended_bytes(
-    // need to slice out the chain code from the private key
-    Buffer.from(v2Key.slice(0, 128), 'hex'),
-  )
+export const getShelleyTxFee = async (
+  IOs: InputOutput,
+  shift: boolean,
+): Promise<BigNumber> => {
+  const out = await getTxOutputTotal(IOs, false)
+  const ins = await getTxInputTotal(IOs, false)
+  const result = ins.minus(out)
+  if (shift) {
+    return result.shiftedBy(-CONFIG.DECIMAL_PLACES_IN_ADA)
+  }
+  return result
 }
-export const v2PkKeyToV3Key = async (v2Key: HdWallet.XPub): PublicKey => {
-  return await PublicKey.from_bytes(
-    // need to slice out the chain code from the public key
-    Buffer.from(v2Key.slice(0, 64), 'hex'),
+
+export const getShelleyTxReceivers = async (
+  signRequest: BaseSignRequest<InputOutput>,
+  includeChange: boolean,
+): Promise<Array<string>> => {
+  const receivers: Array<string> = []
+
+  const changeAddrs = new Set(
+    signRequest.changeAddr.map((change) => change.address),
   )
+  const outputs = await signRequest.unsignedTx.outputs()
+  for (let i = 0; i < (await outputs.size()); i++) {
+    const output = await outputs.get(i)
+    const addr = Buffer.from(
+      await (await output.address()).as_bytes(),
+    ).toString('hex')
+    if (!includeChange) {
+      if (changeAddrs.has(addr)) {
+        continue
+      }
+    }
+    receivers.push(addr)
+  }
+  return receivers
+}
+
+export const shelleyTxEqual = async (
```

Comment:
is this actually used somewhere? if it’s, also I would change the name to `IsShelleyTxEqual` or something like that

Reply:
it comes from Yoroi-Frontend where we need to compare if two txs for some functionality that doesn't exist in Yoroi Mobile I think

Reply:
Good point. We don't use, I ported this code from frontend and thought we might eventually need it as well. I'm removing it for now.